include/trieste/ast.h

```diff
@@ -762,6 +762,12 @@ namespace trieste
       return out.str();
     }
 
+    size_t hash()
+    {
+      // TODO: Implement something that doesn't go via a string
```

Comment:
Is this blocking for merge?

Reply:
My feeling is that it is not blocking. Going via a string is slower, but I think we prefer slower fuzz testing on unique trees over fast fuzz testing on the same tree. I could remove the TODO and add it as an issue instead.

Reply:
I removed the TODO and implemented a hash function that does not go via string :)